front/app/component-library/components/Table/Table.tsx

```diff
@@ -48,6 +49,19 @@ const StyledBox = styled(Box)<{ innerBorders?: InnerBorders }>`
       border-bottom: none;
     }
   `}
+
+  ${({ innerBorders }) =>
+    !innerBorders?.bodyCells
+      ? ''
+      : `
+    tbody > tr > td {
+      border-right: 1px solid ${colors.grey200};
+    }
+
+    tbody > tr > td:last-child {
+      border-right: none;
+    }
+  `}
```

Comment:
Minor: I think the positive check is easier to read but I'll leave that to you

```suggestion
${({ innerBorders }) =>
  innerBorders?.bodyCells
    ? `
  tbody > tr > td {
    border-right: 1px solid ${colors.grey200};
  }
  tbody > tr > td:last-child {
    border-right: none;
  }
`
    : ''}
```